### PR TITLE
Return the DeviceWebToken as part of CreateSessionResponse

### PR DESCRIPTION
**Pull request by codingllama**
Return the DeviceWebToken created by Auth (#38704) all the way from the Proxy to the Web UI.

https://github.com/gravitational/teleport.e/issues/3236

### CODE REVIEW

File: lib/web/apiserver_login_test.go [hunk 1]
```diff
@@ -337,6 +346,158 @@ func TestAuthenticate_rateLimiting(t *testing.T) {
...
+	clock          clockwork.Clock
+	user, password string
+	otpSecret      string // base32-encoded shared OTP secret
+}
+
+// loginWebOTP logins the user using the /webapi/sessions/new endpoint.
+//
+// This is a lower-level utility for tests that want access to the returned
+// CreateSessionResponse.
+func loginWebOTP(t *testing.T, ctx context.Context, params loginWebOTPParams) *CreateSessionResponse {
```
Comment (codingllama) [inline lib/web/apiserver_login_test.go:423]: This and loginWebMFA were put together by stealing bits from various other tests. We don't really have good, standalone helpers that operate at this level of abstraction (some are tied to webPack, others to WebSuite, various tests roll out their own, etc).

I might follow up tomorrow with some testing refactors so we use these more widely (and substitute the many variants popping around).

Comment (flyinghermit) [reply lib/web/apiserver_login_test.go:423]: Yeah there are fair amount of authenticated user setup in `oss` spread across test cases. 🎖️🫡 if you could come up with a reusable standard-ish test helper.

File: lib/web/apiserver_login_test.go [hunk 2]
```diff
@@ -337,6 +346,158 @@ func TestAuthenticate_rateLimiting(t *testing.T) {
 	}
 }
 
+func TestAuthenticate_deviceWebToken(t *testing.T) {
+	pack := newWebPack(t, 1 /* numProxies */)
+	authServer := pack.server.AuthServer.AuthServer
+	proxy := pack.proxies[0]
+	clock := pack.clock
+
+	// Mimic a successfully create DeviceWebToken, regarldess of any parameters.
```
Comment (flyinghermit) [inline lib/web/apiserver_login_test.go:355]:
```suggestion
	// Mimic a valid DeviceWebToken, regardless of any parameters.
```

**Review by flyinghermit — APPROVED** [review]

**Review by zmb3 — APPROVED** [review]

File: lib/web/apiserver_login_test.go (hunk 2 above)
Comment (codingllama) [reply lib/web/apiserver_login_test.go]: This godoc was all kinds of terrible, thanks for spotting it.

File: lib/web/apiserver_login_test.go (hunk 1 above)
Comment (codingllama) [reply lib/web/apiserver_login_test.go:423]: It is, unsurprisingly, kinda hard to unentangle. I have a draft PR branched from this that I started yesterday. I think I'll have to change the function signatures a bit and maybe provide a few "overloaded" variants, but it's cutting a bit of the repetition.

I'll CC you when I mail it.

Comment (codingllama) [reply lib/web/apiserver_login_test.go:423]: Oh, lib/web only. I'm not brave enough to tackle _all_ of OSS.

File: lib/web/apiserver.go [hunk 3]
```diff
@@ -2085,6 +2085,11 @@ type CreateSessionResponse struct {
 	// a user WebUI session can be left idle before being logged out
 	// by the server. A zero value means there is no idle timeout set.
 	SessionInactiveTimeoutMS int `json:"sessionInactiveTimeout"`
+	// DeviceWebToken is the token used to perform on-behalf-of device
```
[the remark this answers is not in the document]
Comment (codingllama) [reply lib/web/apiserver.go:2088]: I'll keep for consistency, as I have it written like this all over the place now. Thanks for pointing it out.

**Comment by codingllama** [comment]
Thanks for the quick review, folks!

**Pull request merged**